hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java

```diff
@@ -1455,6 +1458,9 @@ public void testRenameFile() throws Exception {
     assertTrue("Renamed failed: /dir/file1", getFs().exists(file1Destin));
     FileStatus[] fStatus = getFs().listStatus(dirPath);
     assertEquals("Renamed failed", 1, fStatus.length);
+
+    // We cannot delete a non-empty bucket.
+    getFs().delete(dirPath, true);
```

Comment:
We can delete a non-empty bucket if recursive is set to true via Filesystem, there is already a check on this [line ](https://github.com/apache/ozone/blob/4ff630082b39ca0ebbb9080913f52f9a7b9d7d18/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java#L1089)for non recursive bucket deletion, I guess we can remove this and other occurrences of this in testRenameFileToDir & testRenameToParentDir,  The test added in TestObjectStoreWithFSO should be good. For recursive bucket deletion I can add separate test in HDDS-6414

Reply:
It seems like this is currently needed for the FSO variant of this test (since recursive behaviour is not being followed correctly for FSO). I will add a check for the bucket layout and delete these directories in the FSO case.
We can remove that condition once the behaviour is fixed.
Does this approach sound good?

Reply:
Yes you are right, I just realised that the test was passing earlier because of incorrect isBucketEmpty() implementation for FSO, please revert the change, We can correct these in HDDS-6414